{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOConnectionError"}],"languages":["swift"],"platforms":["macOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/nio\/nioconnectionerror"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/org.swift.docc\/s5ErrorP","doc:\/\/org.swift.docc\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"identifier":{"url":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"A representation of all the errors that happened during an attempt to connect"},{"type":"text","text":" "},{"type":"text","text":"to a given host and port."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOConnectionError"}],"title":"NIOConnectionError","roleHeading":"Structure","role":"symbol","symbolKind":"struct","externalID":"s:3NIO18NIOConnectionErrorV","modules":[{"name":"NIO"}],"navigatorTitle":[{"kind":"identifier","text":"NIOConnectionError"}]},"hierarchy":{"paths":[["doc:\/\/org.swift.docc\/documentation\/NIO"]]},"topicSections":[{"title":"Instance Properties","identifiers":["doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/connectionErrors","doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/dnsAAAAError","doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/dnsAError","doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/host","doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/port"]}],"references":{"doc://org.swift.docc/documentation/NIO/NIOConnectionError/dnsAError":{"role":"symbol","title":"dnsAError","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"dnsAError"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?"}],"abstract":[{"type":"text","text":"The error we encountered doing the DNS A lookup, if any."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/dnsAError","kind":"symbol","type":"topic","url":"\/documentation\/nio\/nioconnectionerror\/dnsaerror"},"doc://org.swift.docc/documentation/NIO/NIOConnectionError/connectionErrors":{"role":"symbol","title":"connectionErrors","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"connectionErrors"},{"kind":"text","text":": ["},{"kind":"typeIdentifier","text":"SingleConnectionFailure","preciseIdentifier":"s:3NIO23SingleConnectionFailureV"},{"kind":"text","text":"]"}],"abstract":[{"type":"text","text":"The errors we encountered during the connection attempts."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/connectionErrors","kind":"symbol","type":"topic","url":"\/documentation\/nio\/nioconnectionerror\/connectionerrors"},"doc://org.swift.docc/documentation/NIO":{"role":"collection","title":"NIO","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO","kind":"symbol","type":"topic","url":"\/documentation\/nio"},"doc://org.swift.docc/documentation/NIO/NIOConnectionError/port":{"role":"symbol","title":"port","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"port"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"}],"abstract":[{"type":"text","text":"The port SwiftNIO was trying to connect to."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/port","kind":"symbol","type":"topic","url":"\/documentation\/nio\/nioconnectionerror\/port"},"doc://org.swift.docc/documentation/NIO/NIOConnectionError":{"role":"symbol","title":"NIOConnectionError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOConnectionError"}],"abstract":[{"type":"text","text":"A representation of all the errors that happened during an attempt to connect"},{"type":"text","text":" "},{"type":"text","text":"to a given host and port."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NIOConnectionError"}],"url":"\/documentation\/nio\/nioconnectionerror"},"doc://org.swift.docc/documentation/NIO/NIOConnectionError/dnsAAAAError":{"role":"symbol","title":"dnsAAAAError","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"dnsAAAAError"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?"}],"abstract":[{"type":"text","text":"The error we encountered doing the DNS AAAA lookup, if any."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/dnsAAAAError","kind":"symbol","type":"topic","url":"\/documentation\/nio\/nioconnectionerror\/dnsaaaaerror"},"doc://org.swift.docc/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/org.swift.docc\/s5ErrorP"},"doc://org.swift.docc/documentation/NIO/NIOConnectionError/host":{"role":"symbol","title":"host","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"host"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[{"type":"text","text":"The hostname SwiftNIO was trying to connect to."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/NIOConnectionError\/host","kind":"symbol","type":"topic","url":"\/documentation\/nio\/nioconnectionerror\/host"},"doc://org.swift.docc/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/org.swift.docc\/s8SendableP"}}}